statsmodels/tsa/vecm/vecm.py

```diff
@@ -0,0 +1,2146 @@
+        Deterministic terms inside the cointegration relation.
+    k_ar : int
+        Lags in the VAR representation. This implies: Lags in the VEC
+        representation = k_ar - 1
+    coint_rank : int
+        Cointegration rank, equals the rank of the matrix :math:`\\Pi` and the
+        number of columns of :math:`\\alpha` and :math:`\\beta`.
+    alpha : ndarray (neqs x coint_rank)
+        Estimate for the parameter :math:`\\alpha` of a VECM.
+    beta : ndarray (neqs x coint_rank)
+        Estimate for the parameter :math:`\\beta` of a VECM.
+    gamma : ndarray (neqs x neqs*(k_ar-1))
+        Array containing the estimates of the p-1 parameter matrices
+        :math:`\\Gamma_1, \\dots, \\Gamma_{p-1}` of a VECM(p-1). The
+        submatrices are stacked horizontally from left to right.
+    sigma_u : ndarray (neqs x neqs)
+        Estimate of white noise process covariance matrix :math:`\\Sigma_u`.
+    deterministic : str {"nc", "co", "ci", "lo", "li"}, default: "nc"
+        * "nc" - no deterministic terms
+        * "co" - constant outside the cointegration relation
+        * "ci" - constant within the cointegration relation
+        * "lo" - linear trend outside the cointegration relation
+        * "li" - linear trend within the cointegration relation
+
+        Combinations of these are possible (e.g. "cili" or "colo" for linear
+        trend with intercept)
+    seasons : int, default: 0
+        Number of seasons. 0 means no seasons.
+    first_season : int, default: 0
+        Season of the first observation.
+    delta_y_1_T : ndarray or None, default: None
+        Auxilliary array for internal computations. It will be calculated if
+        not given as parameter.
+    y_min1 : ndarray or None, default: None
+        Auxilliary array for internal computations. It will be calculated if
+        not given as parameter.
+    delta_x : ndarray or None, default: None
+        Auxilliary array for internal computations. It will be calculated if
+        not given as parameter.
+    model : VECM
+        An instance of the VECM class.
+    names : list of str
+        Each str in the list represents the name of a variable of the time
+        series.
+    dates : array-like
+        For example a DatetimeIndex of length nobs_tot.
+
+    Returns
+    -------
+    **Attributes**
+    nobs : int
+        Number of observations (excluding the presample).
+    model : see Parameters
+    y_all : see endog_tot in Parameters
+    exog : see Parameters
+    exog_coint : see Parameters
+    names : see Parameters
+    dates : see Parameters
+    neqs : int
+        Number of variables in the time series.
+    k_ar : see Parameters
+    deterministic : see Parameters
+    seasons : see Parameters
+    first_season : see Parameters
+    alpha : see Parameters
+    beta : see Parameters
+    gamma : see Parameters
+    sigma_u : see Parameters
+    det_coef_coint : ndarray (#(determinist. terms inside the coint. rel.) x r)
```

Comment:
params_xxx

IIRC, we discussed somewhere about using a Params class so we can keep all the different components of params together.
right now I think it's enough if all parameters are clearly identifiable as such,  `params_xxx`